src/main/java/xyz/jpenilla/squaremap/client/util/WorldTime.java

```diff
@@ -0,0 +1,54 @@
+package xyz.jpenilla.squaremap.client.util;
+
+import net.minecraft.world.level.Level;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+
+public class WorldTime {
+    public static WorldTime instance = new WorldTime();
+    private final Calendar calendar;
+
+    public static WorldTime getInstance() {
+        return instance;
+    }
+
+    private String time;
+    private long lastChecked = 0;
+
+    private WorldTime() {
+        this.calendar = new GregorianCalendar();
+    }
+
+    public String getWorldTime(Level level, boolean realTime, boolean use24Hrs) {
+        long now = System.currentTimeMillis();
+        if (now - this.lastChecked > 500) {
```

Comment:
might want to increase the update frequency so time updates "smoothly" when using world time